### PR TITLE
STL-1779 | Removed extended footer from Nutzungprüfen & Steuerformula…

### PR DESCRIPTION
**Pull request by VictorDelCampo**
# Short Description
- Part of the footer, what I call the extended footer, is now removed from:
   -  Eligibility steps
   -  Tax declaration form steps (as of declaration of further incomes)

# Changes
- The footer template has now a new if-statement that checks the variable disable_extended_footer. This variable is set in two different ways, depending on the step:
   - if the step has been already reactified, an additional parameter is passed to the render_react_component function.
   - if the step has not been reactified, the addional parameter is set in the render_info
- Every react-component has in the corresponding python file this extra variable.
- For those steps without react component I tried to set the additional parameter in the render_info in an upper class, checking that subclasses with steps that should retain the extended footer are not affected. There are 2 special cases:
  -  FormStep is used by 4 steps that should retain the extended footer, but those 4 steps have been already reactified and therefore setting the additional "disable_extended_footer" parameter in the render_info won't affect the display of the extended footer
  - DisplaySteuerlotseStep is only used by 4 eligibility steps and therefore shouldn't be a problem.
- I updated all cypress tests, so that always an extra step (defined within the commands.py) checks if the extended footer is displayed or not.

# Feedback
- Is the if-statement in the footer template ok? All other steps/components without the variable "disable_extended_footer", neither in the render_info not as an additional parameter will still display the extended footer
- Too much changes in the cypress tests?

# How to review this Pull Request
[Link to Confluence](https://digitalservice4germany.atlassian.net/wiki/spaces/STL/pages/117637157/Development+Process#Approve-or-comment) (internal)

### CODE REVIEW

File: webapp/app/forms/steps/lotse/lotse_step.py [hunk 1]
```diff
@@ -21,7 +21,7 @@ def prepare_render_info(cls, stored_data, input_data=None, should_update_data=Fa
 
     def _main_handle(self):
         super()._main_handle()
-
+        self.render_info.additional_info['disable_extended_footer'] = True
```
[the remark this answers is not in the document]
Comment (SannyNguyenHung) [reply webapp/app/forms/steps/lotse/lotse_step.py:24]: It's fine for me :) 

We discussed this and both came to the conclusion that no matter how we implement it, there may not be an elegant alternative that could respond well to any modifications.

**Review by SannyNguyenHung — APPROVED** [review]
LGTM!

File: webapp/app/forms/steps/lotse/lotse_step.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (JulianRoesner) [reply webapp/app/forms/steps/lotse/lotse_step.py:24]: If we would have it as a class attribute, we would at least be able to turn it off and on for the whole Lotse flow. WDYT?

File: webapp/client/cypress/integration/FahrtkostenpauschalePersonA.spec.js [hunk 2]
```diff
@@ -6,6 +6,7 @@ describe("FahrtkostenpauschalePersonA", () => {
   context("with no data", () => {
     beforeEach(() => {
       cy.visit("/lotse/step/person_a_requests_fahrtkostenpauschale");
+      cy.extended_footer_is_disabled(true);
```
Comment (JulianRoesner) [inline webapp/client/cypress/integration/FahrtkostenpauschalePersonA.spec.js:9]: Why do you test this for all calls to cy.visit?

Couldn't we just test it as a unit test for the different pages that when we set the correct attribute that we then also set no footer?
However, we might need to test in the backend that we set the correct flag.

I am fine either way but I was just a little bit confused that we test this part so much while we know that it only depends on the flag.

File: webapp/client/cypress/integration/login.spec.js [hunk 3]
```diff
@@ -49,7 +50,8 @@ describe("Login", () => {
   context("when logged in", () => {
     beforeEach(() => {
       cy.login();
-      cy.visit("/lotse/step/start");
+      cy.visit("/");
```
Comment (JulianRoesner) [inline webapp/client/cypress/integration/login.spec.js:53]: Why did you change the route here?

File: webapp/client/cypress/support/commands.js [hunk 4]
```diff
@@ -79,3 +79,17 @@ Cypress.Commands.add("register", (idnr, dob) => {
...
+    name: "extended_footer_is_disabled",
+  });
+  cy.get("footer")
+    .children()
+    .should("have.length", disabled ? 1 : 2);
+  cy.get("span")
+    .contains("Hilfe und Kontakt")
+    .should(disabled ? "not.exist" : "exist");
+  cy.get("a").contains("Datenschutzerklärung").should("exist");
+});
```
Comment (JulianRoesner) [inline webapp/client/cypress/support/commands.js:95]: 👍

File: webapp/client/cypress/integration/FahrtkostenpauschalePersonA.spec.js (hunk 2 above)
Comment (VictorDelCampo) [reply webapp/client/cypress/integration/FahrtkostenpauschalePersonA.spec.js:9]: I had this thought as well after I changed all the existing tests and you are actually right. We added a new functionality --> disabling the "extended" footer and therefore we should create a new test that checks all that pages without the extended footer and also the other pages which retain it. I will change it

File: webapp/client/cypress/integration/login.spec.js (hunk 3 above)
Comment (VictorDelCampo) [reply webapp/client/cypress/integration/login.spec.js:53]: Because that page doesn't have the extended footer anymore and therefore we can't test if the links are disabled or not. But going to the start page after logging in should keep the same scope of "I am logged in and I can't cancel my unlock code", wdyt?

Comment (JulianRoesner) [reply webapp/client/cypress/integration/login.spec.js:53]: Yes, you are right. I did not see the tests and we can test that functionality also on the front-page 👍

File: webapp/app/forms/steps/lotse/lotse_step.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (VictorDelCampo) [reply webapp/app/forms/steps/lotse/lotse_step.py:24]: Sounds good! But we would also need it for the FormStep class, right?

[the remark this answers is not in the document]
Comment (JulianRoesner) [reply webapp/app/forms/steps/lotse/lotse_step.py]: Yes. I would set it in the FormStep class and the overwrite it for the LotseFormStep?

File: webapp/app/forms/steps/lotse/lotse_step.py [hunk 5]
```diff
@@ -7,6 +7,7 @@
 class LotseFormSteuerlotseStep(FormSteuerlotseStep):
     template = 'basis/form_standard.html'
     header_title = None
+    disable_extended_footer = True
```
Comment (JulianRoesner) [inline webapp/app/forms/steps/lotse/lotse_step.py:10]: Why not set this for the SteuerlotseStep with False and then set it to True here?

File: webapp/client/cypress/integration/extendedFooter.spec.js [hunk 6]
```diff
@@ -0,0 +1,327 @@
...
+  });
+
+  context("unlock code activation page", () => {
+    it("Should show extended footer", () => {
+      cy.visit("/unlock_code_activation/step/data_input");
+    });
+  });
+});
+
+describe("Extended footer is not shown", () => {
```
Comment (JulianRoesner) [inline webapp/client/cypress/integration/extendedFooter.spec.js:79]: Just a small remark: Because we test the disabled property, I find it easier to read if the test description matches what we test 🤷 
```suggestion
describe("Extended footer is disabled", () => {
```

**Review by JulianRoesner — APPROVED** [review]
I like the changes. Two small remarks. 🚀

File: webapp/client/cypress/integration/extendedFooter.spec.js (hunk 6 above)
Comment (VictorDelCampo) [reply webapp/client/cypress/integration/extendedFooter.spec.js:79]: Done

File: webapp/app/forms/steps/lotse/lotse_step.py (hunk 5 above)
Comment (VictorDelCampo) [reply webapp/app/forms/steps/lotse/lotse_step.py:10]: Done
